lib/ansible/modules/system/ufw.py

```diff
@@ -353,17 +381,23 @@ def ufw_version():
             if (ufw_major == 0 and ufw_minor >= 35) or ufw_major > 0:
                 cmd.append([params['comment'], "comment '%s'" % params['comment']])
 
+            # delete rules with --dry-run doesn't display "Rules updated" messages
+            # we compare ## tupple
+            cmd.append([module.check_mode and module.boolean(params['delete']), "| grep -E '^### tuple'"])
```

Comment:
This won't work. The command is not executed by a shell, so `ufw` will get the command line argument `"| grep -E '^### tuple'"` (as a single argument value).

Reply:
Have you got any idea how to simulate a grep in python or with ansible lib ?

Reply:
`output = ''.join([line for line in output.splitlines(keepends=True) where line.startswith('### tuple')])`

That will split the string `output` into lines (keeping the line endings), take only the lines which start with `### tuple`, and concatenate these lines together.